builtin/logical/ssh/util.go

```diff
@@ -135,3 +148,134 @@ func substQuery(tpl string, data map[string]string) string {
 
 	return tpl
 }
+
+// keys parses the input parameters and returns the public and private keys
+// by either generating them or using the provided ones.
+func (b *backend) keys(data *framework.FieldData) (string, string, error) {
```

Comment:
Could we name this a little more descriptively? I was expecting this to be storage-level but fetching but was confused by the comment initially that it was about the parameter parsing.

This was moved from `path_config_ca` if I'm not mistaken?

Reply:
Yes, most of it is the code that was in "Submit CA Information" endpoint handler.